### PR TITLE
jd-gui: 1.4.0 -> 1.6.5

### PR DESCRIPTION
**Pull request by averelld**
###### Motivation for this change
Automatic update fails, upstream updated the build process, and got rid of a gradle 2 dependency in the process.
According to https://github.com/NixOS/nixpkgs/pull/58528#issuecomment-478302487 this could mean gradle_2 can be removed (And probably 3 as well).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/tools/security/jd-gui/default.nix [hunk 1]
```diff
@@ -59,42 +52,37 @@ let
...
-    cp build/libs/${name}.jar ${jar}
-    cp src/linux/resources/jd_icon_128.png $out/share/icons/hicolor/128x128/apps/jd-gui.png
+    cp ${jar} ${outJar}
+    cp ${src}/src/linux/resources/jd_icon_128.png $out/share/icons/hicolor/128x128/apps/jd-gui.png
 
     cat > $out/bin/jd-gui <<EOF
     #!${runtimeShell}
     export JAVA_HOME=${jre}
-    ${jre}/bin/java -jar ${jar} $@
+    ${jre}/bin/java -jar ${outJar} \$@
```
Comment (liff) [inline pkgs/tools/security/jd-gui/default.nix]: Could use `exec` here and quote `"$@"`:

```bash
exec ${jre}/bin/java -jar ${outJar} "\$@"
```

**Comment by liff** [comment]
Built and tested, works as expected.

File: pkgs/tools/security/jd-gui/default.nix [hunk 2]
```diff
@@ -1,56 +1,49 @@
...
+      export JAVA_HOME=${jdk}
+      export GRADLE_USER_HOME=.
+      ./gradlew --no-daemon --init-script ${gradleInit} jar
+    '';
+
+    installPhase = "cp build/libs/${pname}-${version}.jar $out";
+
+    outputHashAlgo = "sha256";
+    outputHashMode = "flat";
+    outputHash = "16wh7cmkhi305i3x3l8qkmcy9z9l67q3rqaxdsy96gih9gbmmzb5";
```
Comment (Mic92) [inline pkgs/tools/security/jd-gui/default.nix]: This seems a bit flacky. Does it not break the checksum, when a new javac changes the way the byte code is compiled?

Comment (averelld) [reply pkgs/tools/security/jd-gui/default.nix]: Yes, unfortunately. Everything is back to the old approach.

**Pull request merged**